core/field.js

```diff
@@ -52,6 +52,52 @@ Blockly.Field = function(text, opt_validator) {
   this.setValidator(opt_validator);
 };
 
+/**
+ * The set of all registered fields, keyed by field type as used in the JSON
+ * definition of a block.
+ * @private
```

Comment:
Add a type annotation.  I think it's:
`@type {string, !{fromJson: Function}}`

Reply:
I'm not that familiar with closure type system, I mostly work in Typescript, but from the docs... https://github.com/google/closure-compiler/wiki/A-word-about-the-type-Object#maps

I think its...

    @type {!Object<string, !{fromJson: Function}>}

I'll add this in

but I was wondering if the value type of the map should define that it inherits from Field, but i don't even know if that's possible with closure, imaginary example:

    @type {!Object<string, constructor extending Blockly.Field>}